Jasmine/Grid/ViewModels/GridViewModel.swift

```diff
@@ -77,63 +80,7 @@ class GridViewModel: GridViewModelProtocol {
 
     /// Returns true iff the game is won
     private var hasGameWon: Bool {
-        let sortedGrid = gridData
-            .sorted { $0.key < $1.key }
-            .map { $0.value }
-
-        // SortedGrid is an array containing all the elements in order.
-        // This will iterate through the array and combine them into Constant.Game.Grid.columns elements,
-        // then check if the resulting string formed is part of the answers.
-        var temporary: [String] = []
-        for element in sortedGrid {
-            temporary.append(element)
-            if temporary.count == Constants.Game.Grid.columns {
-                if answers.contains(where: { $0 == temporary }) {
-                    temporary = []
-                } else {
-                    return false
-                }
-            }
-        }
-        return true
-    }
-
-    /// Loads the grid from an array of array of characters.
-    /// The characters will be shuffled before putting it in the grid.
-    /// Each element in the main array should have the same length.
-    ///
-    /// - Parameter characters: the characters to be loaded to the grid
-    private func loadGrid(from phrases: [[String]]) {
-        let rows = phrases.count
-        let cols = phrases.first?.count ?? 0
-        assert(rows > 0 && cols > 0, "Number of rows and columns should be more than 0")
-        assert(phrases.map { $0.count }.isAllSame, "All rows should have the same length")
-
-        let allChars = phrases.joined().shuffled()
-
-        // Place back allChars to the grid
-        gridData.removeAll()
-        var idx = 0
-        for row in 0..<rows {
-            for col in 0..<cols {
-                gridData[Coordinate(row: row, col: col)] = allChars[idx]
-                idx += 1
-            }
-        }
-
-        delegate?.redisplayAllTiles()
-    }
-
-    /// Swaps two coordinates in the grid data. The built-in swap function cannot be used as
-    /// it will introduce concurrent write in gridData's didSet listener.
-    ///
-    /// - Parameters:
-    ///   - coord1: the first coordinate
-    ///   - coord2: the second coordinate
-    private func swapGridData(_ coord1: Coordinate, and coord2: Coordinate) {
-        let initialSecond = gridData[coord2]
-        gridData[coord2] = gridData[coord1]
-        gridData[coord1] = initialSecond
+        return gridData.allRowsInside(stringArrays: answers)
```

Comment:
Returns true if and only if all rows in `gridData` is contained in `answers`. Can you suggest a better name? Haha also not sure from here.